src/chapters/angular-wk1/templates.rst

```diff
@@ -0,0 +1,176 @@
+         </div>
+      </div>
+      <hr>
+      <div class="links">
+         <h2>Links</h2>
+         <a href="https://www.launchcode.org/" target="_blank">LaunchCode</a> <br>
+         <a href="https://www.webelements.com/" target="_blank">WebElements</a>
+      </div>
+   </body>
+
+We could drastically improve the appearance and content of the page by playing
+around with the tags, classes, styles and text. However, any change we want to
+make needs to be coded directly into the HTML and CSS files.
+
+This quickly becomes inefficient, especially if changing the items needs to
+occur in multiple blocks of code.
+
+A Better Way
+^^^^^^^^^^^^^
+
+Each section in a template contains one or more "blanks" where specific items
+need to be added. Separate JavaScript code sends data to the template to fill
+in these blanks, and this data can change based on a user's actions.
+
+.. sourcecode:: html
+   :linenos:
+
+   <body>
+      <h1>{{mainHeading}}</h1>
+      <div class="row list">
+         <div class='movie'>
+            <h4>Movies</h4>
+            <ul>{{movieTitles}}</ul>
+            <button class='movie'>More</button>
+         </div>
+         <div class='school'>
+            <h4>Education</h4>
+            <ul>{{schoolNames}}</ul>
+            <button class='school'>More</button>
+         </div>
+         <div class='hobby'>
+            <h4>Hobbies</h4>
+            <ul>{{hobbies}}</ul>
+            <button class='hobby'>More</button>
+         </div>
+      </div>
+      <hr>
+      <div class="links">{{Heading + linkList}}</div>
+   </body>
+
+This HTML looks similar to the previous example, but saves about 16 lines. It
+provides the same ``<div></div>`` structure but replaces some of the specific
+text between the tags with *placeholders*.
+
+Each item listed inside ``{{}}`` refers to data that will be passed into the
+template and automatically formatted. For example, the template converts
+``{{movieTitles}}`` into a sequence of ``<li></li>`` tags.
+
+By defining our template in an even more general manner, we could replace the
+``h4``, ``ul`` and ``button`` structure with a single placeholder.
+
+.. sourcecode:: html
+   :linenos:
+
+   <body>
+      <h1>{{mainHeading}}</h1>
+      <div class="row list">
+         <div class='movie'>{{movieContent}}</div>
+         <div class='school'>{{schoolContent}}</div>
+         <div class='hobby'>{{hobbyContent}}</div>
+      </div>
+      <hr>
+      <div class="links">{{linkContent}}</div>
+   </body>
+
+By using a template to build the website, changing the list of movies, schools,
+or hobbies just involves altering something as simple as an array or object.
+After changing the data, the template does the tedious work of modifying the
+HTML. The list of movies would update automatically if we add "Up" to our
+``favoriteMovies`` array, which then gets passed into ``{{movieContent}}``. We
+do not need to worry about re-coding any of the tags.
```

Comment:
For this entire section: A Better Way -- it sings the praises of reducing the total lines of code, and making our code more simple to work with, which is true. However, it doesn't mention the power of templating when working with dynamic data. Students have worked with fetch() and JSON already so should vaguely be aware of getting data from an API. Can you somehow work in a section about not knowing what our data is before we make a fetch request? In that case you have to use a template because we literally don't know what the content is.

Reply:
Done.  I added a new section to address this.  Let me know what you think.